source/run-command/tests/unified/runCommand.yml

```diff
@@ -87,6 +87,9 @@ tests:
               commandName: ping
 
   - description: attaches the provided $readPreference to given command
+    runOnRequirements:
+      # $readPreference is not sent to standalone servers
+      topologies: [ replicaset, sharded-replicaset, load-balanced, sharded ]
```

Comment:
[NODE-5207](https://jira.mongodb.org/browse/NODE-5207) was already resolved, so I'm not sure how this was missed unless you're either not testing on standalones or sending `$readPreference` to standalones. I believe the latter conflicts with the [Server Selection spec](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#topology-type-single):

> Type Standalone: clients MUST NOT send the read preference to the server

I see the [Run Command spec](https://github.com/mongodb/specifications/blob/master/source/run-command/run-command.rst#readpreference) states:

> If the provided ReadPreference is NOT {mode: primary}, the command sent MUST include the $readPreference global command argument.

This sentence follows a reference to the Server Selection spec, but read on its own one could argue it conflicts. In contrast, [Global Command Arguments](https://github.com/mongodb/specifications/blob/master/source/message/OP_MSG.rst#global-command-arguments) in the OP_MSG spec defers to the Server Selection spec for rules on appending `$readPreference`.

Reply:
The Node driver is sending $readPreference to standalones